### ISSUE TITLE
Annotation values that are default are not visible in KSP

### ISSUE THREAD

**Issue by yigit** [google/ksp#53]
When reading a Java annotation with default value, its value is not visible in the arguments list.

```
public @interface JavaAnnotation {
    String debug();
    String withDefaultValue()  default "";
}
```

If you annotate an element with `JavaAnnotation` and don't specify a `withDefaultValue` property, KSP processor threats it as not existing such that arguments map in KSAnnotation will only include `debug` field.

**Comment by yigit** [comment]
i think this bug might already be fixed?
https://android-review.googlesource.com/c/platform/frameworks/support/+/1513790 is passing.

**Comment by yigit** [comment]
actually nevermind, looks like it is still broken for defaults values w/ other annotations.

**Comment by neetopia** [comment]
Yes, it is broken for annotations declared in kotlin and then compiled.

**Comment by neetopia** [comment]
fixed in #528

**Comment by yigit** [comment]
Unfortunateely this still looks broken in beta07.
To reproduce, you can enable [this](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:room/room-compiler-processing/src/test/java/androidx/room/compiler/processing/XAnnotationTest.kt;l=560?q=XAnnotationTest&ss=androidx%2Fplatform%2Fframeworks%2Fsupport) line in the test in AndroidX

**Comment by elihart** [comment]
I think I ran into an edge case in 1.0 where, even with this fix, I get a null default value in the case where a class is defined within the annotation class.

Happens with both kotlin and java - here is a kotlin example.
```
annotation class Example(
    val classParam: KClass<*> = DEFAULT::class
) {
    object DEFAULT
}
```

Moving the object outside the annotation class fixes the problem

**Comment by neetopia** [comment]
is the above class from a compiled library?

**Comment by elihart** [comment]
Yes, the annotation class is defined in a compiled library.

**Comment by neetopia** [comment]
thanks, I'll create a separate issue to track.
